thoth/adviser/steps/security_indicators.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+# thoth-adviser
+# Copyright(C) 2020 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Score based on security indicators aggregated."""
+
+from typing import Any
+from typing import Optional
+from typing import Tuple
+from typing import List
+from typing import Dict
+from typing import TYPE_CHECKING
+import logging
+import os
+
+from thoth.python import PackageVersion
+
+from ..enums import RecommendationType
+from ..exceptions import NotAcceptable
+from ..state import State
+from ..step import Step
+
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SecurityIndicatorStep(Step):
+    """A step that scores a state based on security info aggregated."""
+
+    _JUSTIFICATION_ADDITION = [
+        {"type": "WARNING", "message": ("Package has no gathered information regarding security."),}
+    ]
+
+    # Weights for Confidence
+    HIGH_CONFIDENCE_WEIGHT = float(os.environ.get("SI_STEP_HIGH_CONFIDENCE_WEIGHT", 1))
+    MEDIUM_CONFIDENCE_WEIGHT = float(os.environ.get("SI_STEP_MEDIUM_CONFIDENCE_WEIGHT", 0.5))
+    LOW_CONFIDENCE_WEIGHT = float(os.environ.get("SI_STEP_LOW_CONFIDENCE_WEIGHT", 0.1))
+
+    # Weights for Security
+    HIGH_SEVERITY_WEIGHT = float(os.environ.get("SI_STEP_HIGH_SEVERITY_WEIGHT", 100))
+    MEDIUM_SEVERITY_WEIGHT = float(os.environ.get("SI_STEP_MEDIUM_SEVERITY_WEIGHT", 10))
+    LOW_SEVERITY_WEIGHT = float(os.environ.get("SI_STEP_LOW_SEVERITY_WEIGHT", 1))
+
+    # Overall weight for security rewards
+    SI_REWARD_WEIGHT = float(os.environ.get("SI_STEP_REWARD_WEIGHT", 0.5))
```

Comment:
I was referring here to `Step.CONFIGURATION_DEFAULT` class property which is used when instantiating the step pipeline unit type. It states its default configuration, an example could be the CVE step:

https://github.com/thoth-station/adviser/blob/dd14bf4db7924b8bd231294e600d3b3ff68f9e5d/thoth/adviser/steps/cve.py#L47

By using it, we will capture the unit configuration in the resulting JSON document which is placed on Ceph (for further analyses). It also gives us ability to adjust these defaults if we would configure the pipeline configuration via a JSON input.